### ISSUE TITLE
Implement a pub sub event pattern

### ISSUE THREAD

**Issue by koalyptus** [koalyptus/TableFilter#80]
As mentioned in the [TODO](https://github.com/koalyptus/TableFilter/wiki/TODO), implement a Publish- Subscribe pattern in order to reduce components coupling and cyclomatic complexity due to components interrelation and unclear separation of concerns.
This is a container issue, changes will be released gradually most probably on a component/extension basis.
To avoid as much as possible regressions, from now on there will be a new release channel in ``npm`` called ``next``. This channel will stream the latest changes (ie pub-sub refactoring) and should not be considered as a stable version. Once there is confidence a ``new`` version is mature enough it will be tagged as ``latest`` and will become available via the usual ``npm install`` command.
Those interested in the ``new`` channel will have to use the ``next`` tag: ``install tablefilter@next``.
In short, there is no change in the workflow a part from the fact that users can also install a package from the ``next`` channel.
